### PR TITLE
Healing threat generation

### PR DESCRIPTION
**Pull request by a-ivanov**
- A little bit of typing for spell logic understanding, cleaning code
- Add threat to all creatures which the healing target is attacking now

### CODE REVIEW

File: game/world/managers/objects/spell/SpellEffectHandler.py [hunk 1]
```diff
@@ -42,21 +42,39 @@ def handle_heal(casting_spell, effect, caster, target):
...
+    def _threat_assist(source, target, source_threat: float):
+        if target.in_combat:
+            creature_observers = [attacker for attacker
+                                  in target.attackers.values()
+                                  if not attacker.object_type_mask & ObjectTypeFlags.TYPE_PLAYER]
+            observers_size = len(creature_observers)
+            if observers_size > 0:
+                threat = source_threat / observers_size
+                for creature in creature_observers:
+                    creature.threat_manager.add_threat(source, threat)
```
Comment (Fluglow) [inline game/world/managers/objects/spell/SpellEffectHandler.py:67]: Would this be better as a method of the caster's `UnitManager`, called from `apply_spell_healing`? The same method could also be used by aura application logic later.

Comment (a-ivanov) [reply game/world/managers/objects/spell/SpellEffectHandler.py]: Good point, fixed.

File: game/world/managers/objects/units/creature/ThreatManager.py [hunk 2]
```diff
@@ -14,11 +14,11 @@ class ThreatManager:
     def __init__(self,
                  owner: UnitManager):
         self.owner = owner
-        self.holders: Dict[int, ThreatHolder] = {}
+        self.holders: dict[int, ThreatHolder] = {}
         self.current_holder: Optional[ThreatHolder] = None
 
     def add_threat(self, source: UnitManager, threat: float):
-        if source != self:
+        if source != self and threat > 0.0:
```
Comment (Fluglow) [inline game/world/managers/objects/units/creature/ThreatManager.py:21]: Adding negative threat (`mod_threat` instead?) should be possible for threat reduction effects.

Comment (a-ivanov) [reply game/world/managers/objects/units/creature/ThreatManager.py]: I thing I understood what u mean. Fixed.

**Review by Fluglow — APPROVED** [review]

**Pull request merged**